app/javascript/components/search/StudySearchProvider.js

```diff
@@ -0,0 +1,49 @@
+import React from 'react'
+import { fetchSearch } from 'lib/scp-api'
+import _cloneDeep from 'lodash/cloneDeep'
+import _assign from 'lodash/assign'
+
+const emptySearch = {
+  params: {
+    terms: '',
+    facets: {}
+  },
+  results: [],
+  isLoaded: false,
+  isError: false,
+  updateSearch: () => { throw new Error('You are trying to use this context outside of a Provider container') }
+}
+
+export const StudySearchContext = React.createContext(emptySearch)
+
+export default class StudySearchProvider extends React.Component {
```

Comment:
A function component would be more consistent with other components, and likely more succinct.

Reply:
Yeah, I just spent some time attempting that, and it was beyond my abilities.  Hopefully soon, though!